### PR TITLE
[Model] Add JambaForSequenceClassification model 

### PR DESCRIPTION
**Pull request by yecohn**
[Model] Add support for embedding model JambaClassfication](https://github.com/vllm-project/vllm/pull/10858)

### CODE REVIEW

**Comment by DarkLight1337** [comment]
Thanks for adding this, do you have a HF repo that we can refer to? Don't forget to update the Supported Models page.

File: vllm/worker/pooling_model_runner.py [hunk 1]
```diff
@@ -110,7 +114,8 @@ def execute_model(
                 intermediate_tensors=intermediate_tensors,
                 **MultiModalKwargs.as_kwargs(multi_modal_kwargs,
                                              device=self.device),
-                **cross_enc_kwargs)
+                **cross_enc_kwargs,
+                **seqlen_agnostic_kwargs)
```
Comment (DarkLight1337) [inline vllm/worker/pooling_model_runner.py:118]: Are these changes intentional?

Comment (yecohn) [reply vllm/worker/pooling_model_runner.py:118]: yes, we want to add thoses kwargs as well

Comment (DarkLight1337) [reply vllm/worker/pooling_model_runner.py:118]: cc @tlrmchlsmth

Comment (tlrmchlsmth) [reply vllm/worker/pooling_model_runner.py:118]: Haven't looked at the rest of the PR but this change in particular looks good to me, assuming `finished_requests_ids` and `request_ids_to_seq_ids` are getting plumbed down properly

**Comment by noamgai21** [comment]
We have a test model that can be used to test the functionality of the code:
https://huggingface.co/ai21labs/Jamba-tiny-reward-dev

**Comment by DarkLight1337** [comment]
> We have a test model that can be used to test the functionality of the code: https://huggingface.co/ai21labs/Jamba-tiny-reward-dev

Thanks for this! I'll update the PR accordingly so that the model tests can pass.

**Review by DarkLight1337 — APPROVED** [review]

**Comment by tomeras91** [comment]
@DarkLight1337 - DCO is the only failing check. Can this get a force merge?

**Comment by DarkLight1337** [comment]
Actually, I just realized that this model supposed to be a reward model, not a classification model... should we rename the model?
